File: TypeSharpSample/BusinessObjects/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TypeSharp;

namespace BusinessObjects
{
    /// <summary>
    /// Testing Comment
    /// </summary>
    [TypeSharpClass]
    public class Order
    {

        public Order()
        {
            Products = new List<Product>();
        }

        /// <summary>
        /// Add products to the order
        /// </summary>
        /// <param name="products">The products to add</param>
        /// <param name="group">The group name</param>
        public void AddProducts(List<Product> products, string group) 
        {
            Products.AddRange(products);
        }

        /// <summary>
        /// Gets or sets the products list
        /// </summary>
        public List<Product> Products { get; set; }
    }
}

File: TypeSharpSample/MVCSolution/BusinessObjects/Order.ts
/// <reference path="BusinessObjects.d.ts" />

module BusinessObjects {

	/**
	* Testing Comment
	*/
	export class Order {
	
		/**
		* Gets or sets the products list
		*/
		Products: Product[];
	
		/**
		* Add products to the order
		* @param products The products to add
		* @param group The group name
		*/
		AddProducts(products: Product[], group: string): void {
			/** @todo Implement AddProducts */
			return;
		}
	
	}
	
}